### PR TITLE
update_collection() does not work for models with unique constraints

### LINKED ISSUE #68: update_collection() does not work for models with unique constraints
Child objects that have a unique constraint on a non-PK field cannot be replaced with another object with the same value in the unique field. The add() and edit() methods cause the collection to be flushed before the records that aren't in the new data list are removed resulting in an uncaught integrity exception.

To fix this issue, I think we would need do the remove step first before adding and editing the remaining records. It may also be a good idea to disable flushing for the add() and edit() calls and flush once before returning to avoid possibly accidentally triggering other constraints.

### PR DESCRIPTION
**Pull request by mtbrock**
Fixes #68

### CODE REVIEW

File: keg_elements/db/utils.py [hunk 1]
```diff
@@ -178,18 +178,28 @@ def find_child(self, data):
...
+                to_edit.append((child, record))
             else:
-                child = self.child_cls.add(_commit=False, **record)
-                self.collection.append(child)
+                to_add.append(record)
 
             self.keep_children.add(child)
+
         self.remove_unmodified()
+        session_flush()
```
Comment (nZac) [inline keg_elements/db/utils.py]: Might be good to wrap this entire function in [`might_commit`](https://github.com/level12/keg-elements/blob/master/keg_elements/db/mixins.py#L20-L21) / `might_flush` so the dev can control it.

File: keg_elements/db/utils.py [hunk 2]
```diff
@@ -178,18 +178,28 @@ def find_child(self, data):
     @no_autoflush
     def update(self):
         """Update the objects associated with the entity"""
```
Comment (nZac) [inline keg_elements/db/utils.py]: While we are in this code, can you give this a much better docstring describing what this function actually does.

File: kegel_app/model/entities.py [hunk 3]
```diff
@@ -59,6 +59,35 @@ def testing_create(cls, **kwargs):
...
+                           server_default=sa.text('FALSE'))
+    unique_field = db.Column(db.Integer, unique=True, nullable=False)
+
+    thing_id = sa.Column(sa.Integer, sa.ForeignKey(Thing.id), nullable=False)
+    thing = sa.orm.relationship(lambda: Thing, backref=sa.orm.backref(
+        'other_things',
+        cascade='all, delete-orphan'
+    ))
+
+    _unique_int = 1
```
Comment (nZac) [inline kegel_app/model/entities.py]: Avoid all of this mess and just use `uuid` on a string field.

File: keg_elements/db/utils.py (hunk 1 above)
Comment (mtbrock) [reply keg_elements/db/utils.py:195]: If only works if we flush at this particular moment so `might_flush` won't do it.

File: keg_elements/db/utils.py (hunk 2 above)
Comment (mtbrock) [reply keg_elements/db/utils.py:180]: I updated the docstring of the `CollectionUpdater` class.

**Pull request merged**